Release Jan18

reszelaz opened the pull request:
This is a PR **to master** for the official [Jan18 release](https://github.com/sardana-org/sardana/milestone/3). Final tests (automatic and manual) should be done based on it before accepting the merge.

So, **this is a call for help from @sardana-org/integrators and any other person willing to help** to test this branch ASAP in their respective systems. 

We, at ALBA offer to do the manual tests for OpenSuse11.1 and Windows7 (unless someone else volunteers for doing them).

Notes:
 - The appveyor artifacts from this PR can be used for manual tests.
 - A suggested check-list for the manual tests can be found [here](https://github.com/sardana-org/sardana/blob/develop/doc/how_to_release.md).

teresanunez commented:
Hi, I have made some tests (not using any Taurus GUIs, only with Spock) in our debian8 installation and
I have not found any problem.

reszelaz commented:
For people willing to test this PR, please use Taurus 4.3.0 - it is already [tagged](https://github.com/taurus-org/taurus/releases/tag/4.3.0) in Taurus master. Soon it will be also officially released: PyPI, github, etc.

mrosanes commented:
### Installation
- [ ] Install Sardana from the tar.gz : `pip install <tarball_artifact_URL>`
- [x] Install Sardana by uncompressing tar.gz: by doing:
python setup.py install --install-scripts /.../.../sardana_install/scripts/ --install-lib /.../.../sardana_install/site-packages/

### Create testing environment and run testsuite
- [x] Start Pool demo1. In a console do `Pool demo1`.
- [x] Start MacroServer demo1 and connect to the Pool demo1.
  In another console do: `MacroServer demo1`
- [x] Set MacroServer's MacroPath to point to the macro examples.
  In another IPython console do:
  `PyTango.DeviceProxy('macroserver/demo1/1').put_property({'MacroPath':'<path_to_sardana_installation_dir>/macroserver/macros/examples'})`
- [x] Restart MacroServer e.g. Ctrl+C in the MacroServer's console and
  start it again.
- [ ] Create spock profile demo2. In another console do `spock --profile=demo2`
- [ ] In spock run `sar_demo` macro.
- [ ] Edit `<path_to_sardana_installation_dir>/sardanacustomsettings.py`
  to point to the demo2 door e.g. `UNITTEST_DOOR_NAME = "door/demo2/1"`
- [ ] Run testsuite. In another console do `sardanatestsuite`

### Test Sardana using Spock
- [ ] Test interactive macros from spock e.g. `ask_for_moveable`, `ask_peak`
- [ ] Execute `umvr` macro and verify that the position updates arrives.
- [ ] Test `expconf`:
  1. Configure scan files using expconf set ScanDir to: `/tmp/` and
     ScanFile to: `demo1.h5, demo1.dat`.
  2. Configure online plot to show counters.
  3. Configure snapshot group: with a motor and the `sys/tg_test/1/ampli`
     attribute.
  4. Add the `sys/tg_test/1/double_scalar` attribute to the measurement
     group.
  5. Open online plot.
  6. Set JsonRecorder to true. In spock do `senv JsonRecorder True`
  7. Run step scan and verify if:
     - Records appear in spock output.
     - Records were stored in scan files.
     - Records were plotted on the online plot
- [ ] Run `showscan` and access to the last scan data.
- [ ] Test `edmac`:
  1. Modify existing macro: `ask_peak` and run it to verify that the change
     was applied.
  2. Create new macro in a new macro library:
     `edmac my_macro <path_to_sardana_installation_dir>/macroserver/macros/examples/my_macros.py
     and run it.

### Test Sardana with TaurusGUI

- [ ] Create the GUI using this [guide](https://sourceforge.net/p/sardana/wiki/Howto-GUI_creation)

#### PMTV (PoolMotorTaurusValue)
- [ ] Move motors from the slit panel in absolute and relative modes.
- [ ] Show expert view.
- [ ] Show compact mode.

#### macroexecutor
- [ ] Execute `ascan` macro
- [ ] Pause it in the middel and resume
- [ ] Abort it
- [ ] Add it to favorites
- [ ] Run `lsm` macro
- [ ] Execute `ascan` from favorites
- [ ] Run `lsmac` macro
- [ ] Execute `ascan` from history
- [ ] Edit `dscan` macro in spock yellow line and run it
- [ ] Restart macroexecutor application
- [ ] Run `lsm` from history
- [ ] Run `ascan` from favorites

#### sequencer
- [ ] Add `ascan` macro to the sequence
- [ ] Add `lsct` macro as a `post-acq` hook of `ascan`
- [ ] Add `dscan` macro to the sequence
- [ ] Run the sequence
- [ ] Save sequence to a file
- [ ] Start new sequence
- [ ] Load sequence from a files
- [ ] Run the loaded sequence

#### sardanaeditor
- [ ] Open sardanaeditor with macroserver name as argument.
- [ ] Browse macro libraries and open an existing macro.
- [ ] Edit existing macro and save & apply chaneges.
- [ ] Execute macro to see if changes were aplied.
- [ ] Create a new macro using template.
- [ ] Execute the newly created macro.

mrosanes commented:
Other tests:

When trying to move a motor group it seems that is finding a motor group already existing; (I can always delete the existing motor group), this could be a bug:

<pre>
Door_demo1_1 [139]: mv [[mot53 3] [mot54 7]]
An error occurred while running Macro 'mv([['mot53', '3.0'], ['mot54', '7.0']]) -> 000d2eec-1e34-11e8-bd74-001d0969db5b':
Exception: A MotorGroup with name '_mg_ms_27827_1' already exists
</pre>

mrosanes commented:
Problem in running sardana editor in suse11, because spyderlib 2.1 is required and it does not seem to be available for suse11.

<pre>
sardana-2.4.0-py2.6.egg/sardana/taurus/qt/qtgui/extra_sardana> python sardanaeditor.py
MainThread     WARNING  2018-03-05 08:51:03,796 TaurusRootLogger: Using old sip 4.7.7 (advised >= 4.9)
MainThread     INFO     2018-03-05 08:51:03,805 TaurusRootLogger: Using "PyQt4" for Qt
/usr/lib64/python2.6/site-packages/PyQt4/uic/Compiler/qobjectcreator.py:2: DeprecationWarning: the sets module is deprecated
  from sets import Set as set # Python 2.3 compatibility
Traceback (most recent call last):
  File "sardanaeditor.py", line 41, in <module>
    from taurus.qt.qtgui.editor import TaurusBaseEditor
  File "homelocal/sicilia/mros/taurus_install/site-packages/taurus/qt/qtgui/editor/__init__.py", line 30, in <module>

    
  File "homelocal/sicilia/mros/taurus_install/site-packages/taurus/qt/qtgui/editor/tauruseditor.py", line 42, in <module>
Exception: TaurusEditor needs spyderlib >= 2.1

</pre>

mrosanes commented:
### Installation
- [ ] Install Sardana from the tar.gz : `pip install <tarball_artifact_URL>`

### Create testing environment and run testsuite
- [ ] Start Pool demo2. In a console do `Pool demo2`.
- [ ] Start MacroServer demo2 and connect to the Pool demo2.
  In another console do: `MacroServer demo2`
- [ ] Set MacroServer's MacroPath to point to the macro examples.
  In another IPython console do:
  `PyTango.DeviceProxy('macroserver/demo1/1').put_property({'MacroPath':'<path_to_sardana_installation_dir>/macroserver/macros/examples'})`
- [ ] Restart MacroServer e.g. Ctrl+C in the MacroServer's console and
  start it again.
- [ ] Create spock profile demo2. In another console do `spock --profile=demo2`
- [ ] In spock run `sar_demo` macro.
- [ ] Edit `<path_to_sardana_installation_dir>/sardanacustomsettings.py`
  to point to the demo2 door e.g. `UNITTEST_DOOR_NAME = "door/demo2/1"`
- [ ] Run testsuite. In another console do `sardanatestsuite`

### Test Sardana using Spock
- [ ] Test interactive macros from spock e.g. `ask_for_moveable`, `ask_peak`
- [ ] Execute `umvr` macro and verify that the position updates arrives.
- [ ] Test `expconf`:
  1. Configure scan files using expconf set ScanDir to: `/tmp/` and
     ScanFile to: `demo1.h5, demo1.dat`.
  2. Configure online plot to show counters.
  3. Configure snapshot group: with a motor and the `sys/tg_test/1/ampli`
     attribute.
  4. Add the `sys/tg_test/1/double_scalar` attribute to the measurement
     group.
  5. Open online plot.
  6. Set JsonRecorder to true. In spock do `senv JsonRecorder True`
  7. Run step scan and verify if:
     - Records appear in spock output.
     - Records were stored in scan files.
     - Records were plotted on the online plot
- [ ] Run `showscan` and access to the last scan data.
- [ ] Test `edmac`:
  1. Modify existing macro: `ask_peak` and run it to verify that the change
     was applied.
  2. Create new macro in a new macro library:
     `edmac my_macro <path_to_sardana_installation_dir>/macroserver/macros/examples/my_macros.py
     and run it.

### Test Sardana with TaurusGUI

- [ ] Create the GUI using this [guide](https://sourceforge.net/p/sardana/wiki/Howto-GUI_creation)

#### PMTV (PoolMotorTaurusValue)
- [ ] Move motors from the slit panel in absolute and relative modes.
- [ ] Show expert view.
- [ ] Show compact mode.

#### macroexecutor
- [ ] Execute `ascan` macro
- [ ] Pause it in the middel and resume
- [ ] Abort it
- [ ] Add it to favorites
- [ ] Run `lsm` macro
- [ ] Execute `ascan` from favorites
- [ ] Run `lsmac` macro
- [ ] Execute `ascan` from history
- [ ] Edit `dscan` macro in spock yellow line and run it
- [ ] Restart macroexecutor application
- [ ] Run `lsm` from history
- [ ] Run `ascan` from favorites

#### sequencer
- [ ] Add `ascan` macro to the sequence
- [ ] Add `lsct` macro as a `post-acq` hook of `ascan`
- [ ] Add `dscan` macro to the sequence
- [ ] Run the sequence
- [ ] Save sequence to a file
- [ ] Start new sequence
- [ ] Load sequence from a files
- [ ] Run the loaded sequence

#### sardanaeditor
- [ ] Open sardanaeditor with macroserver name as argument.
- [ ] Browse macro libraries and open an existing macro.
- [ ] Edit existing macro and save & apply chaneges.
- [ ] Execute macro to see if changes were aplied.
- [ ] Create a new macro using template.
- [ ] Execute the newly created macro.

daneos commented:
Here are results of my tests on Debian 9.2.1 with Tango 9.2.5a (omniORB 4.2.2, zeromq 4.2.3), PyTango 9.2.2 and Taurus 4.3.0:

1. Can't pass the testsuite. I can see many errors about not found elements (probably PQDN/FQDN issue, I'm using FQDN as TANGO_HOST), and a couple of ZMQ errors. Full output: [testsuite_debian9.txt](https://github.com/sardana-org/sardana/files/1781449/testsuite_debian9.txt)

2. There is a problem with moving multiple axes at once (maybe related to what @mrosanes found).
```
Door_demo2_1 [6]: umv mot01 0 mot02 -344 mot03 753.991
     mot01      mot02      mot03
 1000.0000    -0.0149     0.0000
An error occurred while running Macro 'umv([['mot01', '0.0'], ['mot02', '-344.0'], ['mot03', '753.991']]) -> e9df6bf8-1ffa-11e8-a646-0800277db4d3':
'NoneType' object has no attribute 'getIndex'

Door_demo2_1 [7]: umv mot01 0 mot02 -344 mot03 753.991
     mot01      mot02      mot03
 1000.0000    -0.0149     0.0000
An error occurred while running Macro 'umv([['mot01', '0.0'], ['mot02', '-344.0'], ['mot03', '753.991']]) -> f4d7b1f0-1ffa-11e8-a646-0800277db4d3':
Exception: A MotorGroup with name '_mg_ms_1873_2' already exists
```

3. Online plot doesn't work.
```
MainThread     WARNING  2018-03-04 23:32:17,513 DynamicPlotManager.tango://vdeb9-sardana.zen.w.daneos.com:10000/Door/demo2/1: Cannot create plot for vdeb9-sardana.zen.w.daneos.com:10000/sys/tg_test/1/double_scalar
```
When I remove `sys/tg_test/1/double_scalar` from the measurement group (or plot configuration) no error is shown, but the plot doesn't start either.

4. `MacroPath` must be specified without trailing slash for `edmac` to work. Traceback: 
[edmac_debian9.txt](https://github.com/sardana-org/sardana/files/1781548/edmac_debian9.txt)

5. After stopping a sequence in Sequencer, start button does not become active. New sequence button is not working at all.

6. Sardanaeditor won't start. It doesn't get added to TaurusGUI automatically (I remember it did some time ago), I can't see it in the menu, and when starting manually I get this error: 
[sardanaeditor_debian9.txt](https://github.com/sardana-org/sardana/files/1781565/sardanaeditor_debian9.txt)

cpascual commented:
@mrosanes wrote:
>[-] Records were plotted on the online plot
sys/tg_test/1/double_scalar is not plotted correctly. Warning message appearing in spock:
```
WARNING 2018-03-02 10:11:35,809 DynamicPlotManager.Door/demo1/1: Cannot create plot for controls02:10000/sys/tg_test/1/double_scalar
MainThread WARNING 2018-03-02 10:11:35,809 DynamicPlotManager.Door/demo1/1: Cannot create plot for controls02:10000/sys/tg_test/1/double_scalar
```
@daneos wrote:
> Online plot doesn't work.

This is probably the same thing. I am looking at it and will make a PR when I have something

amilan commented:
Hi, 
I have run the manual tests described in [here](https://github.com/sardana-org/sardana/blob/develop/doc/how_to_release.md), under CentOS 7, PyTango 9.2.0 and Taurus 4.3.0 and so far I haven't found any big issue that prevents us to accept this PR.

Tomorrow I will continue with a couple of small tests, but I think it could be accepted.

reszelaz commented:
I've reviewed the output of manual tests on openSuse 11.1 done by @mrosanes. Below my comments:

> [-] MacroServer finish smoothly if stopping it with Ctrl+C (using PyTango version 8.1.4)

Reported https://github.com/sardana-org/sardana/issues/719. I think it is not release critical.

> [x - ] Records were stored in scan files.
> In hdf5 file, the dataset entry for sys/tg_test/1/double_scalar is saved as 'systg_test1double_scalar'.
> 

This is normal cause we need to sanitize the names to conform to NeXus best-practices for datasets. 

> [-] Records were plotted on the online plot
> sys/tg_test/1/double_scalar is not plotted correctly. Warning message appearing in spock:
> WARNING 2018-03-02 10:11:35,809 DynamicPlotManager.Door/demo1/1: Cannot create plot for controls02:10000/sys/tg_test/1/double_scalar
> MainThread WARNING 2018-03-02 10:11:35,809 DynamicPlotManager.Door/demo1/1: Cannot create plot for controls02:10000/sys/tg_test/1/double_scalar

According to https://github.com/sardana-org/sardana/pull/711#issuecomment-370809890 @cpascual will take care about this.

> [x-] Move motors from the slit panel in absolute and relative modes.
> But: the limit switches 'indicators' in the taurus GUI using PyTango8.1.4, Taurus3.7.5, Sardana2.4.0; appears sometimes in orange even if the limits are not hit.

This was already reported: https://github.com/sardana-org/sardana/issues/111. While reviewing this ticket I decided to close it as _wontfix_. More details in the issue.

> When trying to move a motor group it seems that is finding a motor group already existing; it can be solved by deleting the existing motor group:

I can not reproduce this. For the moment I suggest to not consider this as release critical.

> Problem in running sardana editor in suse11, because spyderlib 2.1 is required and it does not seem to be available for suse11.

There is not too much that can be done here. For our internal needs at ALBA we will not invest time to get it working on openSuse 11.1. An alternative could be to try to make it work on openSuse 12.1 but IMHO we should concentrate on debian 9.

cmft commented:
Hi @daneos,
I can not reproduce this in the debian docker container (PyTango 9.2.0, Tango 9.2.5a, Taurus and Sardana Jan2018 ):

> After stopping a sequence in Sequencer, start button does not become active. New sequence button is not working at all.

cmft commented:
@daneos  wrote:
> Sardanaeditor won't start. It doesn't get added to TaurusGUI automatically (I remember it did some time ago), I can't see it in the menu, and when starting manually I get this error:

The macroeditor was remove some time ago from the taurusgui due to a solved bug. But it works if you run it independently.

```
can't see it in the menu, and when starting manually I get this error:
```
root@taurus-test:/sardana# python -m sardana.taurus.qt.qtgui.extra_sardana.sardanaeditor macroserver/cfalcon/1
MainThread     INFO     2018-03-07 14:24:20,976 TaurusRootLogger: Using "PyQt4" for Qt
MainThread     INFO     2018-03-07 14:24:21,091 taurus.qt.qtgui.icon.icon: Setting Tango icon theme (from /taurus/lib/taurus/qt/qtgui/icon/)
MainThread     WARNING  2018-03-07 14:24:21,225 TaurusRootLogger: /sardana/src/sardana/taurus/qt/qtgui/extra_sardana/measurementgroup.py:37: DeprecationWarning: taurus.qt.qtgui.resource is deprecated since 4.0. Use taurus.qt.qtgui.icon instead
  from taurus.qt.qtgui.resource import getIcon, getThemeIcon

MainThread     WARNING  2018-03-07 14:24:25,380 TaurusRootLogger: /sardana/src/sardana/taurus/core/tango/sardana/macroserver.py:896: DeprecationWarning: _get_value is deprecated since 4.0. Use .rvalue instead
  elems = CodecFactory().decode(evt_value.value, ensure_ascii=True)

MainThread     WARNING  2018-03-07 14:24:25,731 TaurusRootLogger: /sardana/src/sardana/taurus/qt/qtcore/tango/sardana/model.py:83: DeprecationWarning: getIcon(":/tango.png") is deprecated since 4.0. Use Qt.QIcon(":tango.png") instead
  return taurus.qt.qtgui.resource.getIcon(TYPE_MAP.get(t, (None, _TNG))[1])

MainThread     WARNING  2018-03-07 14:26:25,805 TaurusRootLogger: /sardana/src/sardana/taurus/qt/qtcore/tango/sardana/model.py:83: DeprecationWarning: getIcon(":/python-module.png") is deprecated since 4.0. Use Qt.QIcon(":python-module.png") instead
  return taurus.qt.qtgui.resource.getIcon(TYPE_MAP.get(t, (None, _TNG))[1])

MainThread     WARNING  2018-03-07 14:26:27,309 TaurusRootLogger: /sardana/src/sardana/taurus/qt/qtcore/tango/sardana/model.py:83: DeprecationWarning: getIcon(":/class.png") is deprecated since 4.0. Use Qt.QIcon(":class.png") instead
  return taurus.qt.qtgui.resource.getIcon(TYPE_MAP.get(t, (None, _TNG))[1])

```

daneos commented:
I've done the tests on CentOS 7 with Tango 9.2.5a (omniORB 4.2.2, zeromq 4.2.2), PyTango 9.2.2 and Taurus 4.3.0, but keep in mind that all Tango/PyTango related packages are Solaris custom build ones.

1. The testsuite is also not passing, however the errors are a bit different than on Debian: [testsuite_centos7.txt](https://github.com/sardana-org/sardana/files/1789797/testsuite_centos7.txt)

2. Problem with multiple motor movement is **not** present.

3. There's exactly the same problem with online plot as on Debian.

4. `edmac` problem (see #718).

5. After stopping the sequence, start button also does not become active. New sequence button is working without problems.

6. Sardanaeditor shows exactly the same error message as on Debian: [sardanaeditor_centos7.txt](https://github.com/sardana-org/sardana/files/1789824/sardanaeditor_centos7.txt)
I guess it might have something to do with `spyder` version, I use 3.2.7.

reszelaz commented:
Just for your information, I'm currently investigating the problems with the testsuite reported by @daneos. I confirm that these are related to FQDN. I'm very close to have it fixed and hope to successfully finish on Friday (tomorrow I'm out of office). Thanks to all for your efforts!

reszelaz commented:
The problem with the testsuite reported by @daneos should be fixed now. I took a decision of dropping support to PQDN TANGO_HOST when Sardana is used with Taurus4. It will be readed when we successfully integrate #668. Furthermore it is not sure if everything works well in Tango when server uses PQDN TANGO_HOST and the client uses FQDN URI - see https://github.com/tango-controls/TangoTickets/issues/14. Sardana with Taurus3 works both with FQDN and PQDN TANGO_HOST as before. You should be aware that there may be problems if you create your Sardana instance in one way and then start using it in the other way - see #667.

@daneos could you please try the testsuite now? I think that the only test that could fail for you is the one that depends on [python-nxs](https://github.com/nexusformat/python-nxs). You would need to install it manually.

@mrosanes could you please repeat the testsuite on openSuse11.1 and run a couple of ascanct manually?

jkotan commented:
Hi Zibi,

Do you really mean drop support for PQDN (not other way round)?

That would mean we have to update all our databases from PQDN to FQDN which is hardly doable.
Moreover, we have a lot of scripts which also would need to be changed from PQDN to FQDN.

Since debian stretch requires Taurus 4 this change blocks us to switch to stretch.

It does not looks like a MINOR change. Shouldn't be Taurus 4.3 bumped to 5  and sardana 2.4 to 3?
That would allows for better longer support of Taurus 4.1 and Sardana 2.2/2.3 (which are free of PQDN problems).

Best regards,
Jan

reszelaz commented:
> Do you really mean drop support for PQDN (not other way round)?

Just to clarify, this refers to Sardana, not to Taurus. Currently Sardana, together with Taurus 4, works only with PQDN TANGO_HOST configuration. When using FQDN TANGO_HOST configuration the affected part of Sardana are the continuous scans. This was revealed by this release tests. In 7b8ebf7 I reverted some _nasty_ adaptations introduced when we started supporting Taurus4. So now Sardana works correctly when FQDN TANGO_HOST configuration is used. This seemed to me more reasonable since Taurus4 uses FQDN in its models full names. Without implementing #668 it will be very difficult to support both FQDN and PQDN when Sardana is used with Taurus4.

But when resting overnight I saw a dayligh:) There may be a chance to support both quite easilly. I will give it a try now and post the results this morning. If it does not work, there is no problem to delay the release do the things properly - implement #668. Don't worry we will find solution that fits well for all users.

reszelaz commented:
I have already implemented it and it looks promising. So now, always when Sardana is used with Taurus 4, its internal full names are FQDN (even if the TANGO_HOST configuration is PQDN) - this is a big part of #668. For the moment I have not included the scheme name `tango:` in the full name but maybe it is a good moment to do it? For the moment it is uploaded to [reszelaz/release-Jan18](https://github.com/reszelaz/sardana/tree/release-Jan18). All the tests works correctly for both PQDN and FQDN TANGO_HOST configurations. I also run some continuous scans manually and it looks really well.

However there is still one problem. The Sardana instances created with Sardana version prior to 2.4 on hosts with PQDN TANGO_HOST configuration are affected. More preciselly the measurement groups configurations won't be compatible when upgrading to Sardana 2.4 and Taurus 4.3 (as already anticipated in #668). If I understands correctly, currently DESY is planning such upgrade. We at ALBA would be also affected by this - we use PQDN TANGO_HOST configurations. So I will work now on implementing the backwards compatibility layer for the measurement group configurations. I plan to implement it in a way that once you apply the measurement group configuration with Sardana 2.4 it won't be again compatible if you rollback to previous versions.

reszelaz commented:
I went even further and added the scheme information to the Sardana elements full name. This allowed to remove a lot of Taurus3/Taurus4 compatibility code. There are just two parts of the code which I did not adapt - the spock genutils and the pre-scan-snapshot part of the experiment description - TODO.

I have implemented the backwards compatibility layer explained in https://github.com/sardana-org/sardana/pull/711#issuecomment-371793323 logging a warning message that the measurement group configuration has to be re-applied in order to set the proper configuration with the FQDN URI full names. This message is logged for every full name that had to be adapted in a measurement group. What may easily log 20 messages for the same measurement group. I suggest to limit it to one message per measurement group - TODO.

I have also fixed the problem with online plot of the external channels.

Now, all tests pass on Debian 9 both when FQDN TANGO_HOST configuration is used.

When Sardana is used with Taurus 3 the full names are kept without changes. I think that with not too much effort we could implement the same change and make sardana use FQDN URI full names regardless of the Taurus version used underneath. I think that this will make our life easier when debugging problems on different systems. Furthermore, more Taurus3/Taurus4 compatibility code could be removed. This would require retrieving the FQDN for the database host on our own, instead of using the Taurus validators (see methods `get_full_name` in `SardanaDevice` and `Pool` Tango classes). Note that this is not necessary but could be convenient. I mark this decission as TODO.

With all this the #668 is almost completed. The only pending thing is: “Sardana model names in taurusgui XML configuration”. But I’m not even sure if there is anything needed to be changed. @cpascual your experience with TaurusGUI may be helpful here.

Unfortunately I won’t be able to help with the release in the following 13 days - I’m out of office. After addressing all TODOs from this comment and other revealed during the tests reported by @daneos (Debian9), @mrosanes (Windows7) and @amilan (CentOS7)  I suggest to repeat the manual tests on one of the platforms.

cpascual commented:
## Manual tests with debian9 (after changes described in https://github.com/sardana-org/sardana/pull/711#issuecomment-371932946)

### Installation
- [x] Install Sardana from the tar.gz : `pip install <tarball_artifact_URL>`

### Create testing environment and run testsuite
- [x] Start Pool demo2. In a console do `Pool demo2`.
- [x] Start MacroServer demo2 and connect to the Pool demo2.
  In another console do: `MacroServer demo2`
- [x] Set MacroServer's MacroPath to point to the macro examples.
  In another IPython console do:
  `PyTango.DeviceProxy('macroserver/demo2/1').put_property({'MacroPath':'<path_to_sardana_installation_dir>/macroserver/macros/examples'})`
- [x] Restart MacroServer e.g. Ctrl+C in the MacroServer's console and
  start it again.
- [x] Create spock profile demo2. In another console do `spock --profile=demo2`
- [x] In spock run `sar_demo` macro.
- [x] Edit `<path_to_sardana_installation_dir>/sardanacustomsettings.py`
  to point to the demo2 door e.g. `UNITTEST_DOOR_NAME = "door/demo2/1"`
- [ ] Run testsuite. In another console do `sardanatestsuite`

**Some unittests failed**
(coming back later)
Strange because in travis it is all ok ...

### Test Sardana using Spock
- [x] Test interactive macros from spock e.g. `ask_for_moveable`, `ask_peak`
- [x] Execute `umvr` macro and verify that the position updates arrives.
- [x] Test `expconf`:
  1. Configure scan files using expconf set ScanDir to: `/tmp/` and
     ScanFile to: `demo1.h5, demo1.dat`.
  2. Configure online plot to show counters: On expconf GUI select for all the counter channels, Plot Type 'Spectrum' and Plot Axes '<mov>'
  3. Configure snapshot group: with a motor and the `sys/tg_test/1/ampli`
     attribute.
  4. Add the `sys/tg_test/1/double_scalar` attribute to the measurement
     group.
  5. Open online plot.
  6. Set JsonRecorder to true. In spock do `senv JsonRecorder True`
  7. Run step scan and verify if:
     - Records appear in spock output.
     - Records were stored in scan files.
     - Records were plotted on the online plot

**NOTE** the first scan froze. After restarting Pool+MS+spock and running again, it all worked fine...

- [x] Run `showscan` and access to the last scan data.

**I stop for today. Tomorrow the following more tests**

- [ ] Test `edmac`:
  1. Modify existing macro: `ask_peak` and run it to verify that the change
     was applied.
  2. Create new macro in a new macro library:
     `edmac my_macro <path_to_sardana_installation_dir>/macroserver/macros/examples/my_macros.py
     and run it.

### Test Sardana with TaurusGUI

- [ ] Create the GUI using this [guide](https://sourceforge.net/p/sardana/wiki/Howto-GUI_creation)

#### PMTV (PoolMotorTaurusValue)
- [ ] Move motors from the slit panel in absolute and relative modes.
- [ ] Show expert view.
- [ ] Show compact mode.

#### macroexecutor
- [ ] Execute `ascan` macro
- [ ] Pause it in the middel and resume
- [ ] Abort it
- [ ] Add it to favorites
- [ ] Run `lsm` macro
- [ ] Execute `ascan` from favorites
- [ ] Run `lsmac` macro
- [ ] Execute `ascan` from history
- [ ] Edit `dscan` macro in spock yellow line and run it
- [ ] Restart macroexecutor application
- [ ] Run `lsm` from history
- [ ] Run `ascan` from favorites

#### sequencer
- [ ] Add `ascan` macro to the sequence
- [ ] Add `lsct` macro as a `post-acq` hook of `ascan`
- [ ] Add `dscan` macro to the sequence
- [ ] Run the sequence
- [ ] Save sequence to a file
- [ ] Start new sequence
- [ ] Load sequence from a files
- [ ] Run the loaded sequence

#### sardanaeditor
- [ ] Open sardanaeditor with macroserver name as argument.
- [ ] Browse macro libraries and open an existing macro.
- [ ] Edit existing macro and save & apply chaneges.
- [ ] Execute macro to see if changes were aplied.
- [ ] Create a new macro using template.
- [ ] Execute the newly created macro.

mrosanes commented:
Testsuite fails in suse11 after last changes of FQDN:
[test_suite_failures.txt](https://github.com/sardana-org/sardana/files/1806146/test_suite_failures.txt)

mrosanes commented:
Tests after last commits related with FQDN
Tested Sardana 64 bits on Suse11 (controls02 host)

### Installation
- [ ] Install Sardana from the tar.gz : `pip install <tarball_artifact_URL>`
- [x] Install Sardana by uncompressing tar.gz: by doing:
python setup.py install --install-scripts /.../.../sardana_install/scripts/ --install-lib /.../.../sardana_install/site-packages/

### Create testing environment and run testsuite
- [x] Start Pool demo2. In a console do `Pool demo2`.
- [x] Start MacroServer demo2 and connect to the Pool demo2.
  In another console do: `MacroServer demo2`
- [x] Set MacroServer's MacroPath to point to the macro examples.
  In another IPython console do:
  `PyTango.DeviceProxy('macroserver/demo2/1').put_property({'MacroPath':'<path_to_sardana_installation_dir>/macroserver/macros/examples'})`
- [x] Restart MacroServer e.g. Ctrl+C in the MacroServer's console and
  start it again.
- [-] MacroServer finish smoothly if stopping it with Ctrl+C (using PyTango version 8.1.4)
- [x] Create spock profile demo2. In another console do `spock --profile=demo2`
- [x] In spock run `sar_demo` macro.
- [x] Edit `<path_to_sardana_installation_dir>/sardanacustomsettings.py`
  to point to the demo2 door e.g. `UNITTEST_DOOR_NAME = "door/demo2/1"`
- [x] Run testsuite. In another console do `sardanatestsuite`

### Test Sardana using Spock
- [x] Test interactive macros from spock e.g. `ask_for_moveable`, `ask_peak`
In general, the interactive macros works correctly on the spock console:
Failure if double Ctrl+C:
https://gist.github.com/mrosanes/0156d79eeb18db1850f3ec8f6782b43d
- [x] Execute `umvr` macro and verify that the position updates arrives.
- [x] Test `expconf`:
  1. Configure scan files using expconf set ScanDir to: `/tmp/` and
     ScanFile to: `demo2.h5, demo2.dat`.
  2. Configure online plot to show counters.
  3. Configure snapshot group: with a motor and the `sys/tg_test/1/ampli`
     attribute.
  4. Add the `sys/tg_test/1/double_scalar` attribute to the measurement
     group.
  5. Open online plot.
  6. Set JsonRecorder to true. In spock do `senv JsonRecorder True`
  7. Run step scan and verify if:
     - [x] Records appear in spock output.
     - [x] Records were stored in scan files: problems in storing in hdf5
In hdf5 file, the dataset entry for sys/tg_test/1/double_scalar is saved as 'systg_test1double_scalar'.
     - [x] Records were plotted on the online plot
- [x] Run `showscan` and access to the last scan data: problems when storing the data in hdf5 format
- [x] Test `edmac`:
  1. [x] Modify existing macro: `ask_peak` and run it to verify that the change
     was applied.
  2. [x] Create new macro in a new macro library:
     `edmac my_macro <path_to_sardana_installation_dir>/macroserver/macros/examples/my_macros.py
     and run it.

### Test Sardana with TaurusGUI

- [x-] Create the GUI using this [guide](https://sourceforge.net/p/sardana/wiki/Howto-GUI_creation)
[-] The instruments are not created with all elements indicated in spock (it is possible to see it from jive)
[-] The panels associated with the instruments do not appear

#### PMTV (PoolMotorTaurusValue)
- [x-] Move motors from the slit panel in absolute and relative modes.
But: the limit switches 'indicators' in the taurus GUI using PyTango8.1.4, Taurus3.7.5, Sardana2.4.0; appears sometimes in orange even if the limits are not hit.
- [x] Show expert view.
- [x] Show compact mode.

#### macroexecutor
- [x] Execute `ascan` macro
- [x] Pause it in the middel and resume
- [x] Abort it
- [x] Add it to favorites
- [x] Run `lsm` macro
- [x] Execute `ascan` from favorites
- [x] Run `lsmac` macro
- [x] Execute `ascan` from history
- [x] Edit `dscan` macro in spock yellow line and run it
- [x] Restart macroexecutor application
- [x] Run `lsm` from history
- [x] Run `ascan` from favorites

#### sequencer
- [x] Add `ascan` macro to the sequence
- [x] Add `lsct` macro as a `post-acq` hook of `ascan`
- [x] Add `dscan` macro to the sequence
- [x] Run the sequence
- [x] Save sequence to a file
- [x] Start new sequence
- [x] Load sequence from a files
- [x] Run the loaded sequence

#### sardanaeditor
Do not apply on Suse11
- Open sardanaeditor with macroserver name as argument.
- Browse macro libraries and open an existing macro.
- Edit existing macro and save & apply chaneges.
- Execute macro to see if changes were aplied.
- Create a new macro using template.
- Execute the newly created macro.

cpascual commented:
I finished my tests. See https://github.com/sardana-org/sardana/pull/711#issuecomment-372349865. 

IMHO there is a release critical bug. see summary in https://github.com/sardana-org/sardana/pull/711#issuecomment-372349865

cmft commented:
> PROBLEM running the scan can fail if the h5 output file has been opened previously in spock with showscan because the file is kept open:
> 
> Traceback (most recent call last):
> (...)
>   File "h5py/h5f.pyx", line 78, in h5py.h5f.open (/build/h5py-b1xqNM/h5py-2.7.0/h5py/h5f.c:2127)
> IOError: Unable to open file (Unable to flock file, errno = 11, error message = 'resource temporarily unavailable')
> Closing the spock session in which showscan was launched fixes this issue
> I seem to remember that this was already reported, but haven't checked yet.

If I don't remember bad  it is due to pymca  issue with Single Writer Multiple Reader (SWMR)

cmft commented:
> PROBLEM The instrument panels created from pool do not show the motors/counter widgets, and errors like this are reported in console:
> 
> MainThread WARNING 2018-03-13 14:40:12,011 TaurusForm: Cannot handle model "tango://tango://taurus-test:10000/expchan/ctctrl01/1". Using default widget.
> 
> Note the duplicated tango:// in the model. This is likely related to the latest changes described in #711 (comment) )
> 
> In order to continue the test, I will use a taurusform where the models have been insterted without the duplication.

This bug is in taurusgui (createInstrumentsFromPool) where was added an extra `tango://` I will create a PR

jkotan commented:
Hi,

in the current Sardana/Taurus Release Jan18 I get still https://github.com/taurus-org/taurus/issues/658
i.e. after starting MacroServer with PQDN settings I get:

    jkotan@jessie:~/sources/sardana$ MacroServer stretch 
    MainThread     INFO     2018-03-13 19:48:57,420 TaurusRootLogger: Using "PyQt4" for Qt
    MainThread     WARNING  2018-03-13 19:48:58,592 TaurusRootLogger: /home/jkotan/testsar/lib/python2.7/site-packages/sardana-2.4.0-py2.7.egg/sardana/taurus/core/tango/sardana/pool.py:1935: DeprecationWarning: _get_value is deprecated since 4.0. Use .rvalue instead
      elems = CodecFactory().decode(evt_value.value, ensure_ascii=True)

    Dummy-1        WARNING  2018-03-13 19:49:10,535 jessie.Speedport_W_724V_Typ_A_05011603_00_003:10000.Pool_stretch_1: Received elements error event API_EventTimeout: Event channel is not responding anymore, maybe the server or event system is down
    Dummy-1        WARNING  2018-03-13 19:49:20,550 jessie.Speedport_W_724V_Typ_A_05011603_00_003:10000.Pool_stretch_1: Received elements error event API_EventTimeout: Event channel is not responding anymore, maybe the server or event system is down
    Dummy-1        WARNING  2018-03-13 19:49:30,561 jessie.Speedport_W_724V_Typ_A_05011603_00_003:10000.Pool_stretch_1: Received elements error event API_EventTimeout: Event channel is not responding anymore, maybe the server or event system is down
    Dummy-1        WARNING  2018-03-13 19:49:40,582 jessie.Speedport_W_724V_Typ_A_05011603_00_003:10000.Pool_stretch_1: Received elements error event API_EventTimeout: Event channel is not responding anymore, maybe the server or event system is down

cpascual commented:
Hi, I just released taurus 4.3.1which contains 2 fixes needed for This release of sardana. It is available on PyPI and as a [github release](https://github.com/taurus-org/taurus/releases/tag/4.3.1)

cpascual commented:
No more release-critical problems pending. Releasing!

Thanks everybody for contributing, testing, reviewing, etc.!
